### PR TITLE
Closes #2771: Only clear Session.searchTerms after request initiated by the user.

### LINKED ISSUE #2771: Clear session searchTerm onLoadRequest
Currently we're clearing the searchTerms onLocationChange, which means that the session only knows about its search terms in the brief moment between onURLCommitted and onLocationChange.

Instead let's copy the implementation we used for focus where we cleared session searchTerms `onLoadRequest` **if and only if** the user interacted (remember, the boolean for this user interaction is incorrectly named on the GV side!)

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 1]
```diff
@@ -302,7 +302,14 @@ class GeckoEngineSession(
                     is InterceptionResponse.Url -> loadUrl(url)
                 }
             }
-            return GeckoResult.fromValue(if (response != null) AllowOrDeny.DENY else AllowOrDeny.ALLOW)
+
+            return if (response != null) {
+                GeckoResult.fromValue(AllowOrDeny.DENY)
+            } else {
+                notifyObservers { onLoadRequest(request.isRedirect) }
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: @pocmo this part I don't understand. Why does `request.isRedirect` mean `triggeredByUserInteraction`.

Comment (pocmo) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: Yeah, I didn't understand this either. But I verified that that's what it actually does (and it says so in the javadoc) and that's how it's used in Focus too. And then filed this yesterday:
https://bugzilla.mozilla.org/show_bug.cgi?id=1545170

Comment (pocmo) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: I'll add this as a comment here (with link to the bug)

Comment (csadilek) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: 👍

**Comment by csadilek** [comment]
@pocmo ah this one has a failing test now.

**Pull request merged**